packages/expo-cli/src/commands/install.js

```diff
@@ -1,28 +1,129 @@
-/**
- * @flow
- */
+/* @flow */
+import * as ConfigUtils from '@expo/config';
+import fs from 'fs';
+import { inflect } from 'inflection';
+import JsonFile from '@expo/json-file';
+import keyBy from 'lodash/keyBy';
+import npmPackageArg from 'npm-package-arg';
+import partition from 'lodash/partition';
+import path from 'path';
+import { Modules } from 'xdl';
 
-import { Android, Simulator } from 'xdl';
+import CommandError from '../CommandError';
+import * as PackageManager from '../PackageManager';
 import log from '../log';
 
-export default (program: any) => {
-  program
-    .command('install:ios')
-    .description('Install the latest version of Expo Client for iOS on the simulator')
-    .asyncAction(async () => {
-      if (await Simulator.upgradeExpoAsync()) {
-        log('Done!');
+async function installAsync(packages, options) {
+  const projectRoot = findProjectRoot(process.cwd());
+  const { exp } = await ConfigUtils.readConfigJsonAsync(projectRoot);
+  const packageManager = PackageManager.createForProject(projectRoot, {
+    npm: options.npm,
+    yarn: options.yarn,
+  });
+
+  if (!fs.existsSync(path.join(exp.nodeModulesPath || projectRoot, 'node_modules'))) {
+    log.warn(`node_modules not found, running ${packageManager.name} install command.`);
+    await packageManager.installAsync();
+  }
+
+  const {
+    compatible,
+    incompatible,
+    dependencies,
+    sdkVersion,
+  } = await listBundledNativeModulesAsync(projectRoot, exp);
+
+  const nativeModules = [];
+  const others = [];
+  const versionedPackages = packages.map(arg => {
+    const spec = npmPackageArg(arg);
+    const { name } = spec;
+    if (
+      ['tag', 'version', 'range'].includes(spec.type) &&
+      name &&
+      (compatible[name] || incompatible[name])
+    ) {
+      if (compatible[name]) {
+        // Unimodule packages from npm registry are modified to use the bundled version.
+        const version = dependencies[name];
```

Comment:
i'm probably reading this wrong, it looks to me like we're depending on the unimodule being in the dependencies of the local copy of the expo package in order to determine the version that should be used. won't they all be removed from dependencies on the expo package?